std/assembly/math.ts

```diff
@@ -1207,9 +1207,8 @@ export namespace NativeMath {
       if (iy >= 0x43400000) yisint = 2;
       else if (iy >= 0x3FF00000) {
         k = (iy >> 20) - 0x3FF;
-        let kcond = k > 20;
-        let offset = select<i32>(52, 20, kcond) - k;
-        let Ly = select<i32>(ly, iy, kcond);
+        let offset = select<u32>(52, 20, k > 20) - k;
+        let Ly = select<u32>(ly, iy, k > 20);
```

Comment:
This seems functionally identical. Could you explain?

Reply:
`Ly` now unsigned which and use unsigned right shift later. Also it little bit help better optimize for binarayen: https://github.com/AssemblyScript/assemblyscript/pull/935/commits/4fb7c8eea95837da06c26009e7b3cd917e51646b#diff-93e27c617730e8cd237eb35e80c78177R3595

Reply:
Still not quite recognizing their fix in our change here.  Since this is obviously problematic, I'd somewhat favor copying their code 1:1 here, also removing the `select`s. Wdyt?

Reply:
I think we could remove `select` now and use ternary operator

Reply:
Unfortunately all selects replaced by branches 😞 in optimized version:
```wat
if (result i32)
  i32.const 52
else
  i32.const 20
end
```
And code size increased

Reply:
That's pretty interesting. This happening in rare cases like that:
https://webassembly.studio/?f=cjr2w6x52ae

But usually `x > y ? 1 : 2` optimize to `select`